src/vm/memory/memory.zig

```diff
@@ -148,9 +184,20 @@ pub const Memory = struct {
         address: Relocatable,
     ) error{MemoryOutOfBounds}!MaybeRelocatable {
         if (address.segment_index < 0) {
-            return self.temp_data.get(address) orelse CairoVMError.MemoryOutOfBounds;
+            if (self.temp_data.get(address) == null) {
+                return CairoVMError.MemoryOutOfBounds;
+            } else {
+                var tempCell = self.temp_data.get(address).?;
+                return tempCell.getValue();
+            }
+        } else {
+            if (self.data.get(address) == null) {
+                return CairoVMError.MemoryOutOfBounds;
+            } else {
+                var cell = self.data.get(address).?;
+                return cell.getValue();
```

Comment:
I think that you can directly do:

```zig
return self.data.get(address).?.maybe_relocatable;
```
no?

Reply:
as above same